### ISSUE TITLE
Generated code with errors for enum

### ISSUE THREAD

**Issue by gibahjoe** [dart-ogurets/dart-openapi-maven#19]
Hello, using the version 3.10 library on this spec, It produces a class `GeocodedWaypoint` with the following property:
`GeocodedWaypointTypesEnum types = GeocodedWaypointTypesEnum.[];`

Spec used

``` json 
{
    "swagger": "2.0",
    "info": {
      "version": "1.0",
      "title": "FooBar"
    },
    "host": "foo.bar.com",
    "basePath": "/dev",
    "tags": [

    ],
    "paths": {
    },
    "securityDefinitions": {
    },
    "definitions": {
      "GeocodedWaypoint": {
        "type": "object",
        "properties": {
          "geocoderStatus": {
            "type": "string",
            "enum": [
              "OK",
              "ZERO_RESULTS"
            ]
          },
          "partialMatch": {
            "type": "boolean"
          },
          "placeId": {
            "type": "string"
          },
          "types": {
            "type": "array",
            "items": {
              "type": "string",
              "enum": [
                "STREET_ADDRESS",
                "STREET_NUMBER"
              ]
            }
          }
        },
        "title": "GeocodedWaypoint"
      }
    }
  }
```

library version:
`3.10`

**Comment by rvowles** [comment]
this is fixed on master, just waiting for some confirmation

**Comment by rvowles** [comment]
resolved in 4.2.
